---
title: Whole Genome Methylation Analysis
author: Genie
type: page
date: 2016-07-22T18:43:56+00:00

---
**Christopher Schröder & Sven Rahmann
  
in collaboration with the Institute of Human Genetics, Essen**

The Chair of Genome Informatics in cooperation with the [Institue of Humen Genetics][1] is part of the &#8216;German Epigenome Programme&#8217; ([DEEP][2]). The project focuses on the analysis of cells connected to complex diseases with high socio-economic impact. DEEP contributes innovative epigenome research data to the ‘International Human Epigenome Consortium’ ([IHEC)][3] by generating and interpreting up to 80 reference epigenomes of selected human cells and tissues.

Our research focus lies on the detection of methylation differences from whole genome bisulphite sequencing (WGBS) and &#8216;Nucleosome Occupancy and Methylome sequencing&#8217; (NOMe-seq). We correlate the differences to other genetic and epigenetic data generated by DEEP e.g. SNPs, long RNA, short RNA, mRNA or ChIP.
  
DEEP is divided into multiple different subprojects with contribution of the Genome Informatics:

<p class="western" lang="en-US">
  The first subproject with major analysis of the UK-Essen dealt with the<strong> epigenetic dynamics of monocyte to macrophage differentiation</strong>, where we have discovered a phagocytic gene network that is repressed by DNA methylation in monocytes and rapidly de-repressed after the onset of macrophage differentiation. The paper was accepted by &#8216;<a href="https://epigeneticsandchromatin.biomedcentral.com/">Epigenetics & Chromatin</a>&#8216;.
</p>

<p class="western" lang="en-US">
  <a href="http://genomeinformatics.uni-due.de/wp-content/uploads/2016/07/epi.png"><img class="aligncenter size-large wp-image-1033" src="http://genomeinformatics.uni-due.de/wp-content/uploads/2016/07/epi-854x1024.png" alt="epi" width="625" height="749" srcset="http://genomeinformatics.uni-due.de/wp-content/uploads/2016/07/epi-854x1024.png 854w, http://genomeinformatics.uni-due.de/wp-content/uploads/2016/07/epi-250x300.png 250w, http://genomeinformatics.uni-due.de/wp-content/uploads/2016/07/epi-768x921.png 768w, http://genomeinformatics.uni-due.de/wp-content/uploads/2016/07/epi-624x748.png 624w, http://genomeinformatics.uni-due.de/wp-content/uploads/2016/07/epi.png 1386w" sizes="(max-width: 625px) 100vw, 625px" /></a>
</p>

<p class="western" lang="en-US">
  S. Wallner, C. Schröder, E. Leitão, T. Berulava, C. Haak et al.. Epigenetic dynamics of<br /> monocyte-to-macrophage differentiation. Epigenetics & Chromatin, 9:33, 2016. ISSN<br /> 1756-8935. doi:10.1186/s13072-016-0079-z.
</p>

<p class="western" lang="en-US">
  During the analysis of the differentiation of monocytes and macrophages, we also observe major differences in the methylation between monocytes of different donors. Our current main focus is directed towards these <strong>interindividual methylation differences</strong>. We correlate the differences with SNPs in more than 1200 samples to show the causation of genetic genotypes and methylation. The project is well advanced and we expect soon a first version of the paper.
</p>

<p class="western" lang="en-US">
  Additional we take part of methylation analysis in other subprojects, for example the <strong>methylation in adipocytes , colon mucosa,</strong> or<strong> uveal melanoma</strong> (not DEEP related).
</p>

 [1]: https://www.uk-essen.de/en/institute00/insthumangenetik0/
 [2]: http://www.deutsches-epigenom-programm.de/
 [3]: http://ihec-epigenomes.org/